engine/access/rpc/backend/backend_transactions.go

```diff
@@ -234,6 +234,8 @@ func (b *backendTransactions) GetTransactionsByBlockID(
 func (b *backendTransactions) GetTransactionResult(
 	ctx context.Context,
 	txID flow.Identifier,
+	blockID flow.Identifier,
+	collectionID flow.Identifier,
```

Comment:
I think the logic in this method needs to be tweaked a bit to ensure that any response matches the search criteria exactly. There seems to be 4 cases:

TxID
* Lookup as it's done today
* Query the EN for the result using the discovered block
* Get the collectionID from the block for the response

TxID, BlockID
* Lookup the block by ID
* Query the EN for the result using the provided block
* Get the collectionID from the block for the response

TxID, CollectionID
* Lookup block by collection ID
* Lookup the collection and make sure the tx exists in it
* Query the EN for the result using the discovered block
* Use the provided collectionID in response

TxID, BlockID, CollectionID
* Lookup the block by ID
* Make sure the block contains the provided collection
* Lookup the collection and make sure the tx exists in it
* Query the EN for the result using the provided block
* Use the provided collectionID in response

In any case where the txID/collectionID/blockID don't match eachother, we could return a `NotFound` with some context about which check failed